### PR TITLE
refactor: package file storage

### PR DESCRIPTION
**Pull request by Mickael-Martos**

### CODE REVIEW

**Comment by floross** [comment]
@Mickael-Martos can you update the title of your PR to match the conventional commit format ?

File: libs/nestjs/file-storage/src/configs/file-storage-configuration.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import {
```
Comment (floross) [inline libs/nestjs/file-storage/src/configs/file-storage-configuration.ts]: Can you rename this file in `file-storage.config.ts` ?

File: libs/nestjs/file-storage/src/interfaces/file-storage-configuration-private.interface.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import { ClientOptions as MinioClientOptions, Region } from 'minio';
+
+import { PresignedDownloadConfiguration } from './presigned-download-configuration.interface';
+import { PresignedUploadConfiguration } from './presigned-upload-configuration.interface';
+
+
+export interface FileStorageConfigurationPrivate extends MinioClientOptions {
```
Comment (floross) [inline libs/nestjs/file-storage/src/interfaces/file-storage-configuration-private.interface.ts]: We should name this `FileStoragePrivateConfiguration` or `FileStoragePrivateConfig` instead

File: libs/nestjs/file-storage/src/interfaces/file-storage-configuration-public.interface.ts [hunk 3]
```diff
@@ -79,10 +79,10 @@ export interface FileStorageConfigurationPublic extends ClientOptions {
   /**
```
Comment (floross) [inline libs/nestjs/file-storage/src/interfaces/file-storage-configuration-public.interface.ts]: We should name this `FileStoragePublicConfiguration` or `FileStoragePublicConfig` instead

File: libs/nestjs/file-storage/src/configs/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export * from './file-storage-configuration';
```
Comment (floross) [inline libs/nestjs/file-storage/src/configs/index.ts]: You missed a newline at the end of file

File: libs/nestjs/file-storage/src/file-storage.module-definition.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { ConfigurableModuleBuilder } from '@nestjs/common';
+
+import { FileStorageConfigurationPublic } from './interfaces';
+
```
Comment (floross) [inline libs/nestjs/file-storage/src/file-storage.module-definition.ts]: You have two newline in a row you can delete one

File: libs/nestjs/file-storage/src/file-storage.module.spec.ts [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+ */
+describe('FileStorageModule', () => {
+  let app: INestApplication;
+  let fileStorageModule: TestingModule;
+
+  const setFileStorageDynamicModule = async (
+    fileStorageConfigurationPublic: FileStorageConfigurationPublic,
+  ) => {
+    // const app: INestApplication;
+    // const fileStorageModule: TestingModule;
```
Comment (floross) [inline libs/nestjs/file-storage/src/file-storage.module.spec.ts]: You should remove this lines if not used

File: libs/nestjs/file-storage/src/file-storage.module.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import { FILE_STORAGE_CONFIGURATION } from './constants';
+import {
+  ConfigurableModuleClass,
+  MODULE_OPTIONS_TOKEN,
+} from './file-storage.module-definition';
+import { FileStorageService } from './services';
+
+@Global()
+@Module({
+  imports: [ConsoleModule],
```
Comment (floross) [inline libs/nestjs/file-storage/src/file-storage.module.ts]: Should not import ConsoleModule, we should get rid of the cli part of this module

File: libs/nestjs/file-storage/src/interfaces/index.ts [hunk 8]
```diff
@@ -1,3 +1,4 @@
 export * from './file-storage-configuration-public.interface';
+export * from './file-storage-configuration-private.interface';
```
Comment (floross) [inline libs/nestjs/file-storage/src/interfaces/index.ts]:
```ts
export * from './file-storage-configuration-public.interface';
export * from './file-storage-configuration-private.interface';
```
Should be renamed:
```ts
export * from './file-storage-public-configuration.interface';
export * from './file-storage-private-configuration.interface';
```

File: libs/nestjs/file-storage/src/services/file-storage.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    expect(fileStorageService).toBeInstanceOf(Client);
+  });
+
+  describe('getPresignedUploadUrl', () => {
+    it('throws an error when file mime type is not allowed', async () => {
+      const fileMimeType =
+        'application/vnd.openxmlformats-officedocument.wordprocessingml.document';
+      const fileSize = 1024;
+
+      // const result =  fileStorageService.getPresignedUploadUrl(fileMimeType, fileSize);
```
Comment (floross) [inline libs/nestjs/file-storage/src/services/file-storage.service.spec.ts]: Should be removed if not used

File: libs/nestjs/file-storage/src/services/file-storage.service.spec.ts [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+  // nest unit test are disabled because they need a minio container
+  // and fake data to run
+
+  // it('should file storage buckets list buckets', async () => {
+  //   const result = await fileStorageService.listBuckets();
+  //   expect(result).toBeInstanceOf(Array);
+  // });
+
+  // it('should generate presigned url to upload new file', async () => {
```
Comment (floross) [inline libs/nestjs/file-storage/src/services/file-storage.service.spec.ts]: Did you finish the tests ?

**Review by floross — CHANGES_REQUESTED** [review]

File: libs/nestjs/file-storage/src/file-storage.module-definition.ts (hunk 5 above)
Comment (Mickael-Martos) [reply libs/nestjs/file-storage/src/file-storage.module-definition.ts]: I have linting issues when removing any of them (need new line between import group and need new line after import if not followed by another import)

**Review by maxmousse — APPROVED** [review]

**Review by floross — APPROVED** [review]

**Pull request merged**